src/psyclone/psyir/nodes/node.py

```diff
@@ -95,9 +95,145 @@ def colored(text, _):
+
+            raise GenerationError(errmsg)
+
+    def append(self, item):
+        ''' Extends list append method with children node validation.
+
+        :param item: item to be appened to the list.
+        :type item: :py:class:`psyclone.psyir.nodes.Node`
+
+        '''
+        self._validate_item(len(self), item)
+        super(ChildrenList, self).append(item)
+
+    def __setitem__(self, index, item):
+        ''' Extends list __setitem__ method with children node validation.
+
+        :param int index: position where to insert the item.
+        :param item: item to be inserted to the list.
+        :type item: :py:class:`psyclone.psyir.nodes.Node`
+
+        '''
+        self._validate_item(index, item)
+        super(ChildrenList, self).__setitem__(index, item)
+
+    def insert(self, index, item):
+        ''' Extends list insert method with children node validation.
+
+        :param int index: position where to insert the item.
+        :param item: item to be inserted to the list.
+        :type item: :py:class:`psyclone.psyir.nodes.Node`
+
+        '''
+        positiveindex = index if index >= 0 else len(self) - index
+        self._validate_item(positiveindex, item)
+        # Check that all displaced items will still in valid positions
+        for position in range(positiveindex, len(self)):
+            self._validate_item(position + 1, self[position])
+        super(ChildrenList, self).insert(index, item)
+
+    def extend(self, items):
+        ''' Extends list extend method with children node validation.
+
+        :param items: list of items to be appened to the list.
+        :type items: list of :py:class:`psyclone.psyir.nodes.Node`
+
+        '''
+        for index, item in enumerate(items):
+            self._validate_item(len(self) + index, item)
+        super(ChildrenList, self).extend(items)
+
+    # Methods below don't insert elements but have the potential to displace
+    # or change the order of the items in-place.
+    def __delitem__(self, index):
+        ''' Extends list __delitem__ method with children node validation.
+
+        :param int index: position where to insert the item.
+
+        '''
+        positiveindex = index if index >= 0 else len(self) - index
+        for position in range(positiveindex + 1, len(self)):
+            self._validate_item(position - 1, self[position])
+        super(ChildrenList, self).__delitem__(index)
+
+    def remove(self, item):
+        ''' Extends list remove method with children node validation.
+
+        :param item: item to be deleted the list.
+        :type item: :py:class:`psyclone.psyir.nodes.Node`
+
+        '''
+        for position in range(self.index(item) + 1, len(self)):
+            self._validate_item(position - 1, self[position])
+        super(ChildrenList, self).remove(item)
+
+    def pop(self, index=-1):
+        ''' Extends list pop method with children node validation.
+
+        :param int index: position where to insert the item.
+
```

Comment:
This returns something? If so, please include a docstring.

Reply:
Done